index.d.ts

```diff
@@ -0,0 +1,50 @@
+type Method =
+  | 'get'
+  | 'delete'
+  | 'head'
+  | 'post'
+  | 'put'
+  | 'patch';
+
+type ResponseType =
+  | 'arraybuffer'
+  | 'blob'
+  | 'json'
+  | 'text'
+  | 'formData';
+
+interface Options extends RequestInit{
+  retry?: number;
+  timeout?: number;
+}
+
+/**
+ * The error has a response property with the Response object.
+ */
+export interface HTTPError extends Error {
+  response: Response;
+}
+
+/**
+ * The error thrown when the request times out.
+ */
+export interface TimeoutError extends Error {
+}
+
+interface Ky {
+	(input: Request | string, options?: Options): Promise<Response>;
+	
+  get (input: Request | string, options?: Options): Promise<Response>;
+  post (input: Request | string, options?: Options): Promise<Response>;
+  put (input: Request | string, options?: Options): Promise<Response>;
+  patch (input: Request | string, options?: Options): Promise<Response>;
+  head (input: Request | string, options?: Options): Promise<Response>;
+  delete (input: Request | string, options?: Options): Promise<Response>;
```

Comment:
There's a lot of duplication on these. Is there any way to reuse the declaration?

Reply:
as far as I know, there is nothing in typescript can reuse these declarations. at least axios declare in this way

Reply:
Sometimes, in simple cases, copy-paste doesn't hurt and is even simpler. I think this is such case when it's totally fine. Easily digestable by readers